CMakeLists.txt

```diff
@@ -1,148 +1,23 @@
-find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Quick Widgets Concurrent REQUIRED)
-
-set(PROJECT_SOURCES
-        src-cpp/main.cpp
-        src-cpp/constants.cpp
-        src-cpp/logs.cpp
-        resources.rcc
-)
-
-if(${QT_VERSION_MAJOR} GREATER_EQUAL 6)
-    qt_add_executable(status-desktop
-        ${PROJECT_SOURCES}
-    )
-else()
-    if(ANDROID)
-        add_library(status-desktop SHARED
-            ${PROJECT_SOURCES}
-        )
-    else()
-        add_executable(status-desktop
-          ${PROJECT_SOURCES}
-        )
-    endif()
-endif()
-
-
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/vendor/status-go bin/status-go)
-
-# The following are dependencies that do not include CMakeList.txt files
-# Create a PR in those projects so we can include them just like it was
-# done with status-go
-include(cmake/QRCodeGen.cmake)
-include(cmake/Keycard.cmake)
-
-# Submodiles
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src-cpp/dotherside)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src-cpp/app)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src-cpp/app_service)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src-cpp/backend)
-
-target_include_directories(status-desktop PUBLIC include)
-
-
-target_compile_definitions(status-desktop
-  PRIVATE $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:QT_QML_DEBUG>)
-
-target_link_libraries(status-desktop
-  PRIVATE
-    Qt${QT_VERSION_MAJOR}::Core
-    Qt${QT_VERSION_MAJOR}::Quick
-    Qt${QT_VERSION_MAJOR}::Widgets
-    DOtherSide
-    statusgo_shared # <- Link status-go with this target (remove `_shared` for static linking)
-    qrcodegen
-    keycard
-    app_service
-    app
-  )
-
-
-add_custom_target(rcc ALL DEPENDS resources.rcc)
-add_custom_command(
-    OUTPUT resources.rcc
-    COMMENT "Building resources.rcc"
-    COMMAND ${CMAKE_COMMAND} -E rm -f ${CMAKE_CURRENT_SOURCE_DIR}/resources.rcc
-    COMMAND ${CMAKE_COMMAND} -E rm -f ${CMAKE_CURRENT_SOURCE_DIR}/ui/resources.qrc
-    COMMAND go run
-      ${CMAKE_CURRENT_SOURCE_DIR}/ui/generate-rcc.go
-      -source=${CMAKE_CURRENT_SOURCE_DIR}/ui
-      -output=${CMAKE_CURRENT_SOURCE_DIR}/ui/resources.qrc
-    COMMAND rcc
-      --binary
-      $<$<CONFIG:Debug>:--no-compress>
-      ${CMAKE_CURRENT_SOURCE_DIR}/ui/resources.qrc
-      ${CMAKE_CURRENT_SOURCE_DIR}/resources/resources.qrc
-      -o ${CMAKE_CURRENT_SOURCE_DIR}/resources.rcc
-    VERBATIM
-    USES_TERMINAL
-)
-
-
-# run
-add_custom_target(run
-  COMMENT "Execute desktop"
-  DEPENDS status-desktop
-  COMMAND  ./status-desktop
-  VERBATIM
-  USES_TERMINAL
-)
-
-
-# run debug
-add_custom_target(run-debug
-  COMMENT "Execute desktop with debugger on port 1234"
-  DEPENDS status-desktop
-  COMMAND  ./status-desktop -qmljsdebugger=port:1234
-  VERBATIM
-  USES_TERMINAL
-)
-
 include(CTest)
-if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME AND BUILD_TESTING)
-    add_subdirectory(test-cpp)
-endif()
+enable_testing()
+
+# status-desktop application
+add_subdirectory(vendor)
+add_subdirectory(libs)
```

Comment:
If we use GLOB and GLOB_RECURSE macro, we won't need CMakeLists for each directory contatining just list sources

Reply:
I like to follow general CMake's usage recommendations not to use globbing but to explicitly list all the sources with their scope based on the developer's intention. I agree that globbing makes code much cleaner. However, splitting into multiple CMakeLists files makes it easier to specify them while sticking to the good behaviour: https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html